pagopa/io-app
# feat(IDPay): [IODPAY-103] InitiativeDetailsScreen requires scrolling

## Short description
introduction of "scrolling is required" logic in `InitiativeDetailsScreen.tsx`


<table>
<tr>
<td>  

https://user-images.githubusercontent.com/60693085/217274189-1a5f07d4-6757-463b-bd7a-75dc35f7ccce.mp4
</td>
<td>

https://user-images.githubusercontent.com/60693085/217274231-6122a27b-393e-4fab-8e0c-8fcb707a96e6.mp4
</td>
</tr>
</table>



## List of changes proposed in this pull request
- continue button requires scrolling if the text overflows
- in that case, it will be grey until the user scrolls to the end of the page 
- in this state, if clicked it will scroll to the bottom and will be pressable next time. [ REMOVED UNTIL FUTURE REDESIGN ]
- conditional render of screen, since during load it is covered by a spinner overlay
- removal of testing-purposes initiativeId header

## How to test
The feature was tested by passing an extremely long ipsum as initiative description, so that the text would always overflow.
Once in the screen the testing procedures that were done are: 
- passing no description to be certain of the correct behaviour in that case
- pressing the continue button before or during any load, which should do nothing
- scrolling to the end and checking the button's change of functionality
- pressing the button after loading and making sure it autoscrolls and updates the button's behaviour
